### ISSUE TITLE
RA able to delete a hierarchy which is used in a list and to re-create it differently generating a failure when publishing or refreshing the lists

### ISSUE THREAD

**Issue by SteeveEbener** [terraframe/geoprism-registry#361]
**List the CGR information**
- CGR version: 0.9.0
- CGR instance (e.g. a URL): Staging

**Describe the bug (clear and concise)**
As a RA you can delete a hierarchy which is already used in a list without error message, resulting in system failure when re-publishing or refreshing the list in question

**To Reproduce**
As RA:
1. Go to the GeoObject & Hiearchies module 
2. Delete one of the hierarchies that is already used in a master list (no error message)
3. Go to the list module
4. Open a list that contains the hierarchy that has been deleted
5. Go to the published section and click the "Publish lists"
6. See that the Jobs fails (first screenshot below)
7. Go to the Exploratory section
8. Open the list and refresh the list
9. See the error message on top (second screenshot on top of the window)

**Expected behavior (clear and concise)**
The RA can't delete a hierarchy that is being used in a list

**Screenshots**
![image](https://user-images.githubusercontent.com/12602163/86501890-ed93fc80-bdcf-11ea-9e8d-8e5c40dcf537.png)

![image](https://user-images.githubusercontent.com/12602163/86501896-fb498200-bdcf-11ea-9856-7d0529bede13.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Google Chrome
 - Version: 83.0.4103.116 (Official Build) (64-bit)

**Additional context (if any)**

**Comment by SteeveEbener** [comment]
It seems that the issue here above might also generating the following error message when accessing the GeoObjects and Hierarchies modules after having implemented the steps:

![image](https://user-images.githubusercontent.com/12602163/86512413-be63a680-be34-11ea-91cd-cbbd3d01e06c.png)

**Comment by rrowlands** [comment]
I took the liberty of also adding checks for existing data when deleting organizations and also external systems.

**Comment by SteeveEbener** [comment]
Thanks Richard,

Does that mean that such a deletion will now be possible only once all the data associated to such hierarchy, organization or external system have been deleted from the graph?

I think this brings us (back?) to the need to discuss if, like users and GeoObject, we should not also have the possibility to inactivate the following instead of deleting them and this to be able to keep the data associated to them in the CGR over a given period of time:
1. An organization
2. A hierarchy 
3. An external system
4. A GeoObject type

This would mean that all of these element should also have a start and end date to make sure that we do not consider the data valid beyond their inactivation. This need is for example capture in ticket #104 for GeoObject types.

**Comment by rrowlands** [comment]
> Does that mean that such a deletion will now be possible only once all the data associated to such hierarchy, organization or external system have been deleted from the graph?

Yes, that is correct

> I think this brings us (back?) to the need to discuss if, like users and GeoObject, we should not also have the possibility to inactivate the following instead of deleting them and this to be able to keep the data associated to them in the CGR over a given period of time:
1. An organization
2. A hierarchy
3. An external system
4. A GeoObject type

> This would mean that all of these element should also have a start and end date to make sure that we do not consider the data valid beyond their inactivation. This need is for example capture in ticket #104 for GeoObject types.

This sounds like a good idea, please raise the issue with Justin or Nathan or create a new ticket for it, as I'm afraid it will be buried in this ticket and be forgotten.

**Comment by SteeveEbener** [comment]
Thanks Richard.

I have captured the discussion points in ticket #371  and I am therefore closing the present ticket
